backend/config/config.go

```diff
@@ -170,8 +176,22 @@ ffbrVM+I91v3R03Svv2Nte2xdbx1RmoI/y3tMyZL
 `
 
 // NewDefaultAppConfig returns the default app config.
-func NewDefaultAppConfig() AppConfig {
+//
+// The localeHint argument indicates user preferred UI language.
+// It can be passed from NativeLocale of backend.Environment as is.
+// If localeHint is invalid, preferred language defaults to English.
+func NewDefaultAppConfig(localeHint string) AppConfig {
+	lang, err := language.Parse(localeHint)
+	// ValueError may be returned when the input is well-formed but the
+	// respective subtag is not recognized, in which case it's ok because
+	// the canonical string representation is still valid.
+	if _, ok := err.(language.ValueError); err != nil && !ok {
+		lang = language.English
+	}
 	return AppConfig{
+		Frontend: map[string]interface{}{
+			"userLanguage": lang.String(),
```

Comment:
The way I understand it, the system langauge is stored in the config on first run, and from then on, that one is used.

Why store the language in the config in the first place? If the user changes the system language, the app won't change the language, right?

Reply:
One case is this. If you pick a different language in the app, close it and open again next day, do you expect it to preserve your language of choice from the day before?

> If the user changes the system language, the app won't change the language, right?

Right. It reads from the config. Config always takes precedence.

Also, I wonder what @thisconnect thinks.

Reply:
> If you pick a different language in the app, close it and open again next day, do you expect it to preserve your language of choice from the day before?

Yes I do. But if I never *explicitly* change the language, I would expect it to be my system language.

Reply:
Right. Very good point and it was like this before. I changed it here, I think. Will try to restore that functionality: never store userLanguage unless it's been manually changed.

Reply:
Done using a "volatile" struct field which is never stored.

Reply:
Replaced "volatile" with `native-locale` backend API endpoint.